wagl/modtran.py

```diff
@@ -704,85 +718,54 @@ def read_spectral_response(fname, as_list=False, spectral_range=None):
-        levels in km.
+    Read a MODTRAN output '*.tp6' ascii file to extract
+    solar zenith angles
+    
+    :param tp6_fname:
+        A 'str' containing the full file pathname of the tp6 
+        data file 
+        
+    :return: 
+        solar zenith angle at surface and top of atmosphere 
+    
     """
-    # define a datatype for the hdr info
-    hdr_dtype = numpy.dtype([('record_length', 'int32'),
-                             ('spectral_unit', 'S1'),
-                             ('relabs', 'S1'),
-                             ('linefeed', 'S1'),
-                             ('mlflx', 'int32'),
-                             ('iv1', 'float32'),
-                             ('band_width', 'float32'),
-                             ('fwhm', 'float32'),
-                             ('ifwhm', 'float32')])
-
-    # datatype for the dataframe containing the flux data
-    flux_dtype = numpy.dtype([('upward_diffuse', 'float64'),
-                              ('downward_diffuse', 'float64'),
-                              ('direct_solar', 'float64')])
-
-    with open(fname, 'rb') as src:
-        # read the hdr record
-        hdr_data = numpy.fromfile(src, hdr_dtype, count=1)
-
-        # maximum flux levels at a spectral grid point
-        levels = hdr_data['mlflx'][0] + 1
-
-        # define a datatype to read a record containing flux data
-        dtype = numpy.dtype([('wavelength', 'float64'),
-                             ('flux_data', 'float64', (levels, 3))])
-
-        # read the rest of the hdr which contains the altitude data
-        altitude = numpy.fromfile(src, dtype='float32', count=levels)
-
-        # read the record length end value
-        _ = numpy.fromfile(src, 'int32', count=1)
-
-        # initialise the FORTRAN read
-        ffile = FortranFile(src)
-
-        # read data from 2600 down to 350
-        flux = {}
-        wavelength_steps = range(2600, 349, -1)
-        for wv in wavelength_steps:
-            data = ffile.read_record(dtype)
-            df = pd.DataFrame(numpy.zeros(levels, dtype=flux_dtype))
-            #df['wavelength'] = data['wavelength'][0]
-            df['upward_diffuse'] = data['flux_data'].squeeze()[:, 0]
-            df['downward_diffuse'] = data['flux_data'].squeeze()[:, 1]
-            df['direct_solar'] = data['flux_data'].squeeze()[:, 2]
-            flux[wv] = df
-
-    # concatenate into a single table
-    flux_data = pd.concat(flux, names=['wavelength', 'level'])
-
-    # setup a dataframe for the altitude
-    altitude = pd.DataFrame({'altitude': altitude})
-    altitude.index.name = 'layer'
-
-    return flux_data, altitude
-
-
-def read_modtran_channel(fname, acquisition, albedo):
+    
+    f = open(tp6_fname,'r')
+    cnt = 0
+    cnt_start = 0
+
+    for line in f.readlines(): 
+        cnt +=1 
+        if fnmatch.fnmatch(line,'*SINGLE SCATTER SOLAR PATH GEOMETRY TABLE FOR MULTIPLE SCATTERING*'):
```

Comment:
Can the tp6 be output in binary?  It might make it a cleaner read from disk. Or even a JSON output?

Reply:
I haven't explored that options yet, as far I can tell, there is not option for  tp6 to be outputted as a binary file